File: matrix2latex.py
import numpy as np

def print_number(f, number):
    if not type(number) == int:
        if number.is_integer():
            number = int(number)
        else:
            number = round(number,
                        -int(np.floor(np.log10(abs(number)))) + 1)
    f.write(str(number))

def scalar2latex(filename, scalar):
    f = open(filename, "w");
    print_number(f, scalar)
    f.close()

def vector2latex(filename, vector):
    f = open(filename, "w");
    s = vector.size
    for i in range(s - 1):
        print_number(f, vector[i])
        f.write('&')
    print_number(f, vector[-1])
    f.close()

def matrix2latex(filename, matrix):
    if not type(matrix) == np.ndarray:
        scalar2latex(filename, matrix)
    elif len(matrix.shape) == 1:
        vector2latex(filename, matrix)
    else:
        f = open(filename, "w");
        r, c = matrix.shape
        for i in range(r):
            for j in range(c - 1):
                print_number(f, matrix[i, j])
                f.write('&')
            print_number(f, matrix[i, -1])
            f.write("\\\\\n")
        f.close()

File: matrix2latex.jl
module Matrix2latex
	export matrix2latex

	significants = 4

	function roundMatrix(matrix)
		out = zeros(size(matrix))
		for i in eachindex(matrix)
			out[i] = signif(matrix[i], significants)
		end
		return out
	end

	function matrix2latex(file::String, matrix::Matrix)
		matrix = roundMatrix(matrix)
		s = size(matrix)
		rows = Vector{String}(s[1])
		for i = 1:s[1]
			rows[i] = join(matrix[i, :], '&')
		end
		stream = open(file, "w")
		join(stream, rows, "\\\\\n")
		close(stream)
	end
	function matrix2latex(file::String, vector::Vector)
		vector = roundMatrix(vector)
		stream = open(file, "w")
		join(stream, vector, '&')
		close(stream)
	end
	function matrix2latex(file::String, number)
		number = signif(number, significants)
		stream = open(file, "w")
		write(stream, string(number))
		close(stream)
	end
end
